resource/store.go

```diff
@@ -244,16 +250,50 @@ func (s *Store) ForceDelete(ctx context.Context, kind string, identifier Identif
 	return err
 }
 
-// List lists all resources of kind in the provided namespace, with optional label filters.
-func (s *Store) List(ctx context.Context, kind string, namespace string, filters ...string) (ListObject, error) {
+// List lists all resources using the Namespace and Filters provided in options. An empty namespace in options is
+// equivalent to NamespaceAll, and an empty or nil Filters slice will be ignored.
+// List will automatically paginate through results, fetching pages based on options.PerPage.
+// To list a single page of results, use ListPage.
+func (s *Store) List(ctx context.Context, kind string, options StoreListOptions) (ListObject, error) {
 	client, err := s.getClient(kind)
 	if err != nil {
 		return nil, err
 	}
-
-	return client.List(ctx, namespace, ListOptions{
-		LabelFilters: filters,
+	resp, err := client.List(ctx, options.Namespace, ListOptions{
+		Limit:        options.PerPage,
+		LabelFilters: options.Filters,
 	})
+	if err != nil {
+		return nil, err
+	}
+	if resp.GetContinue() == "" {
```

Comment:
nit: I think this conditional is redundant - if it's ever empty you'll just skip past the `for` to the end of the function.

Reply:
Holdover from an initial pass where a new list was created and everything was added to that. Will remove this check.